src/library/tests/test_views.py

```diff
@@ -32,35 +32,34 @@
         # Need to do this since the setupClass sets the current tenant connection to
         # cls.tenant. We cannot create the library tenant if we are not using the public schema
         # But check first if the library tenant already exists
+        cls.library_tenant = Tenant.objects.filter(schema_name=get_library_schema_name()).first()
 
-        if not Tenant.objects.filter(schema_name=cls.get_library_schema_name()).exists():
+        if not cls.library_tenant:
             with tenant_context(_public_tenant):
                 cls._setup_library_tenant()
 
     @classmethod
     def get_library_tenant_domain(cls):
-        return f'{cls.get_library_schema_name()}.test.com'
-
-    @classmethod
-    def get_library_schema_name(cls):
-        return f"{apps.get_app_config('library').TENANT_NAME}"
+        return f'{get_library_schema_name()}.test.com'
 
     @classmethod
     def _setup_library_tenant(cls):
         # Setup the library tenant
-        cls.library_tenant = Tenant(schema_name=cls.get_library_schema_name(), name='Library Tenant')
+        cls.library_tenant = Tenant(schema_name=get_library_schema_name(), name='Library Tenant')
         cls.library_tenant.save(verbosity=cls.get_verbosity())
 
         # Setup the domain
         library_domain = cls.get_library_tenant_domain()
         cls.library_domain = TenantDomain(tenant=cls.library_tenant, domain=library_domain)
         cls.library_domain.save()
 
```

Comment:
@yujinyuz is there a reason why you aren't full_clean before the save here?  We've been running into problems in our database because we weren't running full_clean on models before saving.  Seems like a weakness in Django that isn't well documented.

Reply:
@yujinyuz see this issue which describes the problem: https://github.com/bytedeck/bytedeck/issues/1630

Reply:
https://jamescooke.info/djangos-model-save-vs-full_clean.html

Reply:
@yujinyuz I am quite aware of how Django's full clean though but I think it should be used whenever we are accepting user inputs ie via forms or django views that aren't using forms but relying on `request.POST.get('key')` for creating or updating instances.

Here's also a good repository https://github.com/HackSoftware/Django-Styleguide?tab=readme-ov-file#validation---clean-and-full_clean

So, I think it doesn't benefit that much calling `full_clean` within tests.

---

But if you want consistency then I can just update this and move on.

Reply:
I'm sorry, I didn't even realize this was in a test!